### ISSUE TITLE
Problem with S3 storage

### ISSUE THREAD

**Issue by nicolasfara** [knadh/listmonk#101]
When i upload a new file (image for example) no preview is shown.

I inspected the element and i find that the href to the image (S3 bucket URL) is concatenated with base URL of my application.

Is there a problem with application or with my settings?
![Screenshot_20200510_180348](https://user-images.githubusercontent.com/11615611/81504279-a82fe300-92e8-11ea-897a-324d5d020ae1.png)

**Comment by mr-karan** [comment]
@nicolasfara Can you please show your `config.toml` ? I just tested it on my end, couldn't reproduce this.

**Comment by nicolasfara** [comment]
Hi
this is my `config.toml` file:
```
[app]
# Interface and port where the app will run its webserver.
address = "0.0.0.0:9000"

# Public root URL of the listmonk installation that'll be used
# in the messages for linking to images, unsubscribe page etc.
root = "https://osta.nicolasfarabegoli.it"

# (Optional) full URL to the static logo to be displayed on
# user facing view such as the unsubscription page.
# eg: https://mysite.com/images/logo.svg
logo_url = "https://osta.nicolasfarabegoli.it/public/static/logo.png"

# (Optional) full URL to the static favicon to be displayed on
# user facing view such as the unsubscription page.
# eg: https://mysite.com/images/favicon.png
favicon_url = "https://osta.nicolasfarabegoli.it/public/static/favicon.png"

# The default 'from' e-mail for outgoing e-mail campaigns.
from_email = "Osta <osta@mambelli.com>"

# List of e-mail addresses to which admin notifications such as
# import updates, campaign completion, failure etc. should be sent.
# To disable notifications, set an empty list, eg: notify_emails = []
notify_emails = ["nicolas.farabegoli@gmail.com"]

# Maximum concurrent workers that will attempt to send messages
# simultaneously. This should ideally depend on the number of CPUs
# available, and should be based on the maximum number of messages
# a target SMTP server will accept.
concurrency = 5

# Maximum number of messages to be sent out per second per worker.
# If concurrency = 10 and message_rate = 10, then up to 10x10=100 messages
# may be pushed out every second. This, along with concurrency, should be
# tweaked to keep the net messages going out per second under the target
# SMTP's rate limits, if any.
message_rate = 5

# The number of errors (eg: SMTP timeouts while e-mailing) a running
# campaign should tolerate before it is paused for manual
# investigation or intervention. Set to 0 to never pause.
max_send_errors = 1000


[privacy]
# Allow subscribers to unsubscribe from all mailing lists and mark themselves
# as blacklisted?
allow_blacklist = false

# Allow subscribers to export data recorded on them?
allow_export = false

# Items to include in the data export.
# profile            Subscriber's profile including custom attributes
# subscriptions      Subscriber's subscription lists (private list names are masked)
# campaign_views     Campaigns the subscriber has viewed and the view counts
# link_clicks        Links that the subscriber has clicked and the click counts
exportable = ["profile", "subscriptions", "campaign_views", "link_clicks"]

# Allow subscribers to delete themselves from the database?
# This deletes the subscriber and all their subscriptions.
# Their association to campaign views and link clicks are also
# removed while views and click counts remain (with no subscriber
# associated to them) so that stats and analytics aren't affected.
allow_wipe = false


# Database.
[db]
host = "db"
port = 5432
user = "listmonk"
password = "listmonk"
database = "listmonk"
ssl_mode = "disable"

# Maximum active and idle connections to pool.
max_open = 50
max_idle = 10

# SMTP servers.
[smtp]
    [smtp.my0]
        enabled = true
        host = "[HIDE]"
        port = "587"

        # "cram", "plain", or "login". Empty string for no auth.
        auth_protocol = "login"
        username = "[HIDE]"
        password = "[HIDE]"

        # Optional. Inform listmonk on which email format to use
        # {html|plain|both} default: both
        email_format = "both"

        # Optional. Some SMTP servers require a FQDN in the hostname.
        # By default, HELLOs go with "localhost". Set this if a custom
        # hostname should be used.
        hello_hostname = ""

        # Maximum time (milliseconds) to wait per e-mail push.
        send_timeout = 5000

        # Maximum concurrent connections to the SMTP server.
        max_conns = 10

    [smtp.postal]
        enabled = false
        host = "my.smtp.server2"
        port = "25"

        # cram or plain.
        auth_protocol = "plain"
        username = "xxxxx"
        password = ""

        # Optional. Inform listmonk on which email format to use
        # {html|plain|both} default: both
        email_format = "both"

        # Maximum time (milliseconds) to wait per e-mail push.
        send_timeout = 5000

        # Maximum concurrent connections to the SMTP server.
        max_conns = 10


[upload]
# File storage backend. "filesystem" or "s3".
provider = "s3"

    [upload.s3]
        # (Optional). AWS Access Key and Secret Key for the user to access the bucket.
        # Leaving it empty would default to use instance IAM role.
        aws_access_key_id = "[HIDE]"
        aws_secret_access_key = "[HIDE]"

        # AWS Region where S3 bucket is hosted.
        aws_default_region="eu-west-1"

        # Bucket name.
        bucket="osta-mambelli"

        # Path where the files will be stored inside bucket. Empty for root.
        bucket_path=""

        # "private" or "public".
        bucket_type="public"

        # (Optional) Specify TTL (in seconds) for the generated presigned URL.
        # Expiry value is used only if the bucket is private.
        expiry="86400"

    [upload.filesystem]
        # Path to the uploads directory where media will be uploaded.
        upload_path="./uploads"

        # Upload URI that's visible to the outside world.
        # The media uploaded to upload_path will be made available publicly
        # under this URI, for instance, list.yoursite.com/uploads.
        upload_uri = "/uploads"
```
I've marked all sensitive data with `[HIDE]` for obvious security reason.

**Comment by mr-karan** [comment]
Thanks @nicolasfara this is indeed a bug as I confirmed. Will roll out a fix for this pretty soon.

**Comment by nicolasfara** [comment]
@mr-karan Thanks for the support! I'll wait the fix :+1:

**Comment by knadh** [comment]
Fixed in [v0.6.0-alpha](https://github.com/knadh/listmonk/releases/tag/v0.6.0-alpha).
